routes/index.js

```diff
@@ -14,12 +16,36 @@ router.get("/login", function (req, res, next) {
   res.render("login");
 });
 
-router.get('/signup/phone', function (req, res, next) {
-  res.render('signup-phone', { title: '배민 - 전화번호 인증'})
+router.get("/signup/phone", function (req, res, next) {
+  res.render("signup-phone", { title: "배민 - 전화번호 인증" });
 });
 
 router.get("/signup/detail", function (req, res, next) {
-  res.render("signup-detail");
+  const hasSignupError = req.session?.signupError;
+  if (!hasSignupError) {
+    res.render("signup-detail", { error: null });
+    return;
+  }
+
+  req.session.signupError = null;
+  res.render("signup-detail", { error: hasSignupError });
 });
 
+router.post("/signup", async function (req, res, next) {
+  const { email, password, nickname } = req.body;
+
+  const hasAlreadyUserEmail = await db.find({ email });
+
+  if (hasAlreadyUserEmail.length) {
+    req.session.signupError = "이미 사용중인 이메일입니다!";
+    res.redirect("/signup/detail");
+    return;
+  }
+
+  const hashedPassword = bycrypt.hashSync(password, 10);
+
+  await db.insert({ email, hashedPassword, nickname });
```

Comment:
```suggestion
  await db.insert({ email, password: hashedPassword, nickname });
```

#53에서 결정한 user 객체 property key가 일치하지 않습니다.

Reply:
헉쓰! 또 이런 실수를!

훌륭한 캐치 감사드립니다~~